cli/cmd/initnodes.go

```diff
@@ -398,3 +418,35 @@ func gethInit(ctx context.Context, cfg InitConfig, dir string) error {
 
 	return nil
 }
+
+func downloadAndRestoreLatestSnapshot(ctx context.Context, network string, clientName string) error {
+	snapshotArchive := getSnapshotBackupArchive(clientName)
+	bucketName := fmt.Sprintf("omni-%s-snapshots", network)
+	gcpCloudStorageURL := fmt.Sprintf("https://storage.googleapis.com/%s/%s", bucketName, snapshotArchive)
+
+	log.Info(ctx, "Downloading and restoring latest snapshot...", "network", network, "client", clientName)
+	if err := downloadUntarLz4(ctx, gcpCloudStorageURL, clientName); err != nil {
+		return errors.Wrap(err, "download untar lz4 error", "network", network, "client", clientName)
+	}
+
+	return nil
+}
+
+func getSnapshotBackupArchive(clientName string) string {
```

Comment:
no need for a method if it is this simple and only used in one place, just inline it.

Reply:
Done.